src/main/java/com/metsci/laproc/plotting/GraphableDataWithMetrics.java

```diff
@@ -91,19 +79,39 @@ public void setName(String name) {
      * @param m The metric to use for this data
      * @return A set of doubles based on the metrivc's calculations
      */
-    private double[] getDataForMetric(Metric m) {
+    private double[] getDataForMetric(ParametricFunction<T> m) {
         double[] values = new double[this.getSize()];
+        Iterator<T> iter = points.iterator();
```

Comment:
Just as something to note, some people find iterators harder to read. Iterating over the points with an enhanced for loop with a counter to keep track of the values index might be easier to read.